client/src/ui/organisms/placement/index.tsx

```diff
@@ -207,6 +223,13 @@ export function PlacementInner({
 
   const observer = useRef<IntersectionObserver | null>(null);
   const timer = useRef<Timer>({ timeout: null, start: null });
+
+  const sendViewed = useCallback(() => {
```

Comment:
This is all a bit smelly to me, IMO the "proper" (react) approach is to wrap our intersection observer in a hook, so we can do something like:

```js
const isVisible = usePageVisibility();
const node = useRef(); // needs to be assigned
const isIntersecting = useIntersectionObserver(node); // needs to be written
const timer = useRef({ timeout: null, start: null });

useEffect(() => {
  if (isVisible && isIntersecting) {
    if (timer.timeout === null) {
      setTimeout(...)
    }
  } else {
    clearTimeout(...)
  }
}, [isVisible, isIntersecting])
```

This is a lot clearer, and more expressive, than the current approach, which takes quite a while to understand.

Reply:
Yes, take a shot at it. I tried this before. The `node` from a `useRef` doesn't trigger an update in the `useIntersectionObserver`. We can have conversation, but since this reports our impressions it's rather delicate.